internal/runbits/dependencies/changesummary.go

```diff
@@ -22,126 +22,70 @@ const showUpdatedPackages = true
 // installed for that package.
 // `artifacts` is an ArtifactMap containing artifacts in the changeset, and `filter` contains any
 // runtime requirements/artifacts already installed.
-func OutputChangeSummary(out output.Outputer, changeset artifact.ArtifactChangeset, artifacts artifact.Map, filter artifact.Map) {
-	// Determine which package was installed.
-	var addedId *artifact.ArtifactID
-	for _, candidate := range changeset.Added {
-		if !isDependency(candidate.ArtifactID, changeset, artifacts) {
-			if addedId != nil {
-				return // more than two independent packages were added
-			}
-			foundId := candidate.ArtifactID
-			addedId = &foundId // cannot address candidateId as it changes over the loop
+func OutputChangeSummary(out output.Outputer, changeset *buildplan.ArtifactChangeset, alreadyInstalled buildplan.Artifacts) {
+	addedString := []string{}
+	addedLocale := []string{}
+	added := buildplan.Ingredients{}
+	dependencies := buildplan.Ingredients{}
+	directDependencies := buildplan.Ingredients{}
+	for _, a := range changeset.Added {
+		added = append(added, a.Ingredients...)
+		for _, i := range a.Ingredients {
+			addedString = append(addedLocale, fmt.Sprintf("%s@%s", i.Name, i.Version))
+			addedLocale = append(addedLocale, fmt.Sprintf("[ACTIONABLE]%s[/RESET]", addedString))
+			dependencies = append(dependencies, i.RuntimeDependencies(true)...)
+			directDependencies = append(dependencies, i.RuntimeDependencies(false)...)
 		}
 	}
-	if addedId == nil {
-		return // no single, independent package was added
-	}
-	added := artifacts[*addedId]
-	logging.Debug("Determined that runtime update was triggered by adding package '%s/%s'", added.Namespace, added.Name)
 
-	// Determine the package's direct and indirect dependencies.
-	dependencies := buildplan.DependencyMapFor(*addedId, artifacts, filter, showUpdatedPackages)
-	directDependencies := make([]artifact.ArtifactID, 0, len(dependencies))
-	uniqueDependencies := make(map[artifact.ArtifactID]bool)
-	for artifactId, indirectDependencies := range dependencies {
-		directDependencies = append(directDependencies, artifactId)
-		uniqueDependencies[artifactId] = true
-		for _, depId := range indirectDependencies {
-			uniqueDependencies[depId] = true
-		}
-	}
-	sort.SliceStable(directDependencies, func(i, j int) bool {
-		return artifacts[directDependencies[i]].Name < artifacts[directDependencies[j]].Name
-	})
-	hasAdditionalIndirectDependencies := len(directDependencies) < len(uniqueDependencies)
+	dependencies = sliceutils.UniqueByProperty(dependencies, func(i *buildplan.Ingredient) any { return i.IngredientID })
+	directDependencies = sliceutils.UniqueByProperty(directDependencies, func(i *buildplan.Ingredient) any { return i.IngredientID })
+	numIndirect := len(dependencies) - len(directDependencies)
```

Comment:
I think I might've, the locale entry reads:

> direct and indirect dependencies

Reply:
Okay, I mixed this up with the logging message below. The en-us.yaml message is

Installing [ACTIONABLE]{{.V0}}@{{.V1}}[/RESET] includes [ACTIONABLE]{{.V2}}[/RESET] direct dependencies, and a total of [ACTIONABLE]{{.V3}}[/RESET] direct and indirect dependencies.

So the first number is direct dependencies, and the second number is direct and indirect dependencies (total, unique dependencies). We either need to change that message, or change it so that `numIndirect` is not being fed by itself as one of the arguments to that message.

Reply:
Yeah that is indeed pretty confusing.. I've changed it to:

> Installing [ACTIONABLE]{{.V0}}[/RESET] includes [ACTIONABLE]{{.V1}}[/RESET] direct dependencies, and [ACTIONABLE]{{.V2}}[/RESET] indirect dependencies.

And have updated the numbers passed in to match.